libraries/node-core-library/src/FileSystem.ts

```diff
@@ -829,25 +892,98 @@ export class FileSystem {
    * @param linkPath - The path to the link.
    */
   public static getRealPath(linkPath: string): string {
-    return fsx.realpathSync(linkPath);
+    return FileSystem._wrapException(() => {
+      return fsx.realpathSync(linkPath);
+    });
   }
 
   /**
    * An async version of {@link FileSystem.getRealPath}.
    */
   public static getRealPathAsync(linkPath: string): Promise<string> {
-    return fsx.realpath(linkPath);
+    return FileSystem._wrapExceptionAsync(() => {
+      return fsx.realpath(linkPath);
+    });
   }
 
   // ===============
   // UTILITY FUNCTIONS
   // ===============
 
   /**
-   * Returns true if the error provided indicates the file or folder
-   * does not exist.
+   * Returns true if the error provided indicates the file or folder does not exist.
    */
   public static isNotExistError(error: NodeJS.ErrnoException): boolean {
-    return error.code === 'ENOENT' || error.code === 'ENOTDIR';
+    return FileSystem.isFileDoesNotExistError(error) || FileSystem.isFolderDoesNotExistError(error);
+  }
+
+  /**
+   * Returns true if the error provided indicates the file does not exist.
+   */
+  public static isFileDoesNotExistError(error: NodeJS.ErrnoException): boolean {
+    return FileSystem._isErrorType(error) && (error.code === 'ENOENT');
+  }
+
+  /**
+   * Returns true if the error provided indicates the folder does not exist.
+   */
+  public static isFolderDoesNotExistError(error: NodeJS.ErrnoException): boolean {
+    return FileSystem._isErrorType(error) && (error.code === 'ENOTDIR');
+  }
+
+  private static _isErrorType(error: NodeJS.ErrnoException): boolean {
+    if (process.env.NODE_ENV !== 'test' && process.env.JEST_WORKER_ID === undefined) {
+      return error instanceof Error;
```

Comment:
`instanceof NodeJS.ErrnoException` seems like a reasonable test, since `ErrnoException` is a Node.js proprietary concept.

`instanceof Error` seems unreliable given that TypeScript models it as an `interface` and the "instance" concept has changed between ES5 and ES6. 

More fundamentally I don't understand why we are testing `error instanceof Error` instead of `error instanceof ErrnoException` or whatever. (?)

Reply:
`NodeJS.ErrnoException` is an interface, so we can't check its type.

It also seems that Jest somehow messes with `Error` such that the thing that `fs` throws isn't what `Error` is defined as.